File: src/student-ui/students-pages/student-dashboard-page/components/ProgressOverview.js
import React from "react";
import { Card, Box, Typography, LinearProgress } from "@mui/material";
import LocalFireDepartmentIcon from "@mui/icons-material/LocalFireDepartment";

const ProgressOverview = ({
  progress = 0.5,
  minutesToday = 0,
  currentStreak = 0,
}) => (
  <Card
    sx={{
      p: 3,
      mb: 3,
      borderRadius: 3,
      background: (theme) =>
        `linear-gradient(90deg, ${theme.palette.secondary.light}33 0%, ${theme.palette.background.paper} 100%)`,
      boxShadow: 2,
    }}
  >
    <Box
      sx={{
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        mb: 2,
      }}
    >
      <Typography variant="h6" fontWeight={700}>
        Today's Progress
      </Typography>
      {currentStreak > 0 && (
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            bgcolor: "#ff980022",
            borderRadius: 2,
            px: 1.5,
            py: 0.5,
          }}
        >
          <LocalFireDepartmentIcon
            sx={{ color: "#ff9800", fontSize: 18, mr: 0.5 }}
          />
          <Typography variant="body2" fontWeight={700} color="#ff9800">
            {currentStreak} day{currentStreak > 1 ? "s" : ""}
          </Typography>
        </Box>
      )}
    </Box>
    <Box sx={{ mb: 2 }}>
      <LinearProgress
        variant="determinate"
        value={progress * 100}
        sx={{
          height: 10,
          borderRadius: 2,
          bgcolor: "background.default",
          "& .MuiLinearProgress-bar": { bgcolor: "primary.main" },
        }}
      />
    </Box>
    <Box
      sx={{
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
      }}
    >
      <Typography variant="body2" fontWeight={700} color="primary.main">
        {Math.round(progress * 100)}% Complete
      </Typography>
      <Typography variant="body2" color="text.secondary">
        {minutesToday > 1440 ? 1440 : minutesToday} min today
      </Typography>
    </Box>
  </Card>
);

export default ProgressOverview;

File: src/student-ui/to react/lib/screens/home/progress_overview.dart
import 'package:englishfluencyguide/theme/app_theme.dart';
import 'package:flutter/material.dart';
import 'package:englishfluencyguide/l10n/app_localizations.dart';

class ProgressOverview extends StatelessWidget {
  final double progress;
  final int minutesToday;
  final int currentStreak;

  const ProgressOverview({
    Key? key,
    required this.progress,
    required this.minutesToday,
    required this.currentStreak,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final l10n = AppLocalizations.of(context)!;

    return Container(
      padding: const EdgeInsets.all(24),
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(20),
        gradient: LinearGradient(
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
          colors: [
            theme.colorScheme.secondaryContainer,
            theme.colorScheme.surface,
          ],
        ),
        boxShadow: const [
          AppTheme.boxShadow,
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                l10n.todaysProgress,
                style: theme.textTheme.titleLarge?.copyWith(
                  fontWeight: FontWeight.bold,
                ),
              ),
              if (currentStreak > 0)
                Container(
                  padding: const EdgeInsets.symmetric(
                    horizontal: 12,
                    vertical: 6,
                  ),
                  decoration: BoxDecoration(
                    color: Colors.orange.withValues(alpha: 0.1),
                    borderRadius: BorderRadius.circular(20),
                    border: Border.all(
                      color: Colors.orange.withValues(alpha: 0.3),
                    ),
                  ),
                  child: Row(
                    mainAxisSize: MainAxisSize.min,
                    children: [
                      const Icon(
                        Icons.local_fire_department,
                        size: 16,
                        color: Colors.orange,
                      ),
                      const SizedBox(width: 4),
                      Text(
                        '$currentStreak day${currentStreak > 1 ? 's' : ''}',
                        style: theme.textTheme.bodySmall?.copyWith(
                          color: Colors.orange,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ],
                  ),
                ),
            ],
          ),
          const SizedBox(height: 20),
          Stack(
            children: [
              Container(
                height: 8,
                decoration: BoxDecoration(
                  color: theme.colorScheme.surfaceVariant,
                  borderRadius: BorderRadius.circular(4),
                ),
              ),
              AnimatedContainer(
                duration: const Duration(milliseconds: 800),
                height: 8,
                width: MediaQuery.of(context).size.width *
                    0.8 *
                    progress.clamp(0.0, 1.0),
                decoration: BoxDecoration(
                  gradient: LinearGradient(
                    colors: [
                      theme.colorScheme.primary,
                      theme.colorScheme.secondary,
                    ],
                  ),
                  borderRadius: BorderRadius.circular(4),
                ),
              ),
            ],
          ),
          const SizedBox(height: 16),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                '${(progress * 100).toInt()}% Complete',
                style: theme.textTheme.bodyMedium?.copyWith(
                  fontWeight: FontWeight.bold,
                  color: theme.colorScheme.primary,
                ),
              ),
              Text(
                '${minutesToday > 1440 ? 1440 : minutesToday} min today',
                style: theme.textTheme.bodyMedium?.copyWith(
                  color: theme.colorScheme.onSurfaceVariant,
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
